tests/adapters/test_visa.py

```diff
@@ -98,6 +98,14 @@ def test_write_read_all_bytes(adapter):
     assert adapter.read_bytes(-1) == b"SCPI,MOCK,VERSION_1.0\n"
 
 
+def test_write_read_break_on_termchar(adapter):
+    """Test read_bytes breaks or does not break on termchar."""
+    adapter.write("*IDN?")
+    adapter.connection.read_termination = ","
+    assert adapter.read_bytes(-1, break_on_termchar=True) == b"SCPI,"
+    assert adapter.read_bytes(-1, break_on_termchar=False) == b"MOCK,VERSION_1.0\n"
```

Comment:
Can we split this in two tests, where in the second you don't give the `break_on_termchar` argument? That way we would automatically confirm that the default behaviour has not changed.

Reply:
The default behavior (newly introduced in 0.11) has changed, at least according to the implementation.
According to intention and docstring, the behavior remains the same.

The reason I put the second assert into the test is, that after the first assert, the second part of the message remains in the read buffer and the following test fails.

I'll make the default implicit to catch future changes of default behavior. 

Probably I have to use some flush or clear method to do it properly.

Reply:
> The reason I put the second assert into the test is, that after the first assert, the second part of the message remains in the read buffer and the following test fails.

If that happens that is a problem that has to be fixed, not circumvented. We cannot leak state from one test to another, that will lead to weird problems sooner or later.

Is the fixture module scoped? Ill check, I'm just on the phone.

If we change the default behaviour compared to the last release,  and we can't or dont want to avoid thst, that should be documented in the changelog.

Reply:
I added a "read_bytes(-1)" to the fixture, such that the read buffer is emptied after each test.
Each test has its own adapter (I checked the fixture, its the default scope), but somehow pyvisa-sim keeps the state...

I added a changelog entry, that now all adapters do not break at term char. I'd consider the previous behaviour as a bug, not a feature with a transition path.

Reply:
These tests, for example succeed (based on `test_visa.py`):
~~~python
import pytest
from pymeasure.adapters import VISAAdapter

@pytest.fixture
def adapter():
    return VISAAdapter('ASRL2::INSTR', visa_library='@sim',
                       read_termination="\n",
                       timeout=10)


def test_com(adapter):
    adapter.write("*IDN?")
    adapter.connection.read_termination = "x"  # change adapter instance
    assert adapter.connection.read_termination == "x"


def test_com2(adapter):
    assert adapter.connection.read_termination == "\n"  # new adapter instance
    assert adapter.read_bytes(-1) == b"SCPI,MOCK,VERSION_1.0\n"  # old read buffer

Reply:
> I added a changelog entry, that now all adapters do not break at term char. 

Thanks!

> I'd consider the previous behaviour as a bug, not a feature with a transition path.

Fair enough.

Reply:
> Each test has its own adapter (I checked the fixture, its the default scope), but somehow pyvisa-sim keeps the state...

This is messed up. I fear that even if we report a bug it's probably not gonne be addressed, pyvisa-sim is not that well/actively maintained.